js/node_shared/unix.js

```diff
@@ -0,0 +1,369 @@
+const { spawnSync } = require("node:child_process");
```

Comment:
It would be nice to have a top level comment explaining why this file exists, and a TL;DR; of "we want unix_spawn, and abuse FDs to get things to work here"